app/gui/src/controller/searcher.rs

```diff
@@ -49,6 +49,74 @@ pub const ASSIGN_NAMES_FOR_NODES: bool = true;
 /// See also [`Searcher::add_enso_project_entries`].
 const ENSO_PROJECT_SPECIAL_MODULE: &str = "Standard.Base.Enso_Project";
 
+/// Name of the virtual component group in the `Standard.Base` library which contains input
+/// components.
+const INPUT_COMPONENT_GROUP_NAME: &str = "Input";
+
+thread_local! {
+    /// Code snippets of default literal values of text and number type. The snippets are
+    /// documented as code that can be used as input nodes. When converted to [`Component`]s and
+    /// added to the [`component::List`] they allow the users to easily enter literals in code.
+    static LITERAL_INPUT_NODES_SNIPPETS: Vec<Rc<component::HardcodedSnippet>> = [
+        LiteralSnippet {
+            name:               "text input",
+            code:               "\"\"",
+            return_type:        "Standard.Base.Data.Text.Text",
+            documentation:
+                "A text input node.\n\n\
+                An empty text. The value can be edited and used as an input for other nodes."
+            ,
+            icon:               ide_view_component_group::icon::Id::TextInput,
+        },
+        LiteralSnippet {
+            name:               "number input",
+            code:               "0",
+            return_type:        "Standard.Base.Data.Numbers.Number",
+            documentation:
+                "A number input node.\n\n\
+                A zero number. The value can be edited and used as an input for other nodes."
+            ,
+            icon:               ide_view_component_group::icon::Id::NumberInput,
+        },
+    ].into_iter().map(|c| Rc::new(c.try_into().unwrap())).collect_vec();
```

Comment:
Magic. Please refactor to something clear and understandable. This expression has now 21 lines!